### PR TITLE
[feature][api-datasource]add redshift auth mode to get connection by access key

### PR DESCRIPTION
**Pull request by Tianqi-Dotes**
## Purpose of the pull request

add redshift auth mode to get connection by access key
https://github.com/apache/dolphinscheduler/issues/13039

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Comment by caishunfeng** [comment]
@Tianqi-Dotes please update the doc together.

File: dolphinscheduler-ui/src/views/datasource/list/detail.tsx [hunk 1]
```diff
@@ -245,7 +246,11 @@ const DetailModal = defineComponent({
                   v-show={showPort}
                   label={t('datasource.port')}
                   path='port'
-                  show-require-mark
+                  show-require-mark={
+                    showMode && detailForm.mode === 'IAM-accessKey'
+                      ? false
+                      : true
```
Comment (Amy0104) [inline dolphinscheduler-ui/src/views/datasource/list/detail.tsx]: It seems that `!showMode || detailForm.mode !== 'IAM-accessKey'` is better.

**Review by Amy0104 — DISMISSED** [review]
The front end part LGTM.

File: dolphinscheduler-ui/src/views/datasource/list/detail.tsx (hunk 1 above)
Comment (Tianqi-Dotes) [reply dolphinscheduler-ui/src/views/datasource/list/detail.tsx]: done

**Review by Amy0104 — APPROVED** [review]
The front end part LGTM.

**Review by caishunfeng — APPROVED** [review]
+1

**Pull request merged**